addon/src/modifiers/sortable-item.js

```diff
@@ -614,11 +688,19 @@ export default class SortableItemModifier extends Modifier {
 
     this._preventClick();
 
+    // Get sortableItems before making drop end and pass it to update function
+    // This is necessary for direction grid, otherwise ordering is not working correctly (because in sortedItems we check if item isDragged)
+    // Calling this getter here, makes no difference for other directions
+    const sortedItems = this.sortableGroup.sortedItems;
+
     set(this, 'isDragging', false);
     set(this, 'isDropping', true);
 
-    this.sortableGroup.update();
-    transitionPromise.then(() => this._complete());
+    this.sortableGroup.update(sortedItems);
+
+    let allTransitionPromise = this._waitForAllTransitions();
+
+    Promise.all([transitionPromise, allTransitionPromise]).then(() => this._complete());
```

Comment:
`_waitForAllTransitions()` is internally checking if any element is still in transition. Thats important, because setting complete must be success after all items did complete transition (not only like now, dragged element).
Without this check, the complete will fired early and this is causing randomly errors (like we have now got with direction grid)

Reply:
this change was moved to PR #562